qemu-shutdown

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+sleep 60
```

Comment:
This here is the intermediate successor of https://github.com/giantswarm/k8s-kvm/pull/18. I will use the other PR for the next steps. The plan is to go with this for the next version and improve it meanwhile. This here is not perfect but it works and serves our purpose well enough for now. That way we get a first step towards updates out and can iterate further. The idea is to make k8s-kvm a go project and wrap the docker entrypoint magic via os/exec in the first step.

Reply:
Current problem is that the master seems to lose the cluster information with this. How come? Before the master deletion I get this. 

```
$ kubectl get node -o wide
NAME                            STATUS    ROLES     AGE       VERSION   EXTERNAL-IP   OS-IMAGE                                        KERNEL-VERSION   CONTAINER-RUNTIME
master-2z08i-7b887d5b8-j92gr    Ready     master    2m        v1.10.1   <none>        Container Linux by CoreOS 1688.5.3 (Rhyolite)   4.14.32-coreos   docker://17.12.1-ce
worker-dk2r4-6b854d46c7-pxqrq   Ready     <none>    2m        v1.10.1   <none>        Container Linux by CoreOS 1688.5.3 (Rhyolite)   4.14.32-coreos   docker://17.12.1-ce
worker-u7smc-7475cb84b9-cxwhx   Ready     <none>    2m        v1.10.1   <none>        Container Linux by CoreOS 1688.5.3 (Rhyolite)   4.14.32-coreos   docker://17.12.1-ce
```

When I delete the master pod and another comes up I get this. 

```
$ kubectl get node -o wide
NAME                           STATUS    ROLES     AGE       VERSION   EXTERNAL-IP   OS-IMAGE                                        KERNEL-VERSION   CONTAINER-RUNTIME
master-2z08i-7b887d5b8-8g5ld   Ready     master    1m        v1.10.1   <none>        Container Linux by CoreOS 1688.5.3 (Rhyolite)   4.14.32-coreos   docker://17.12.1-ce
```

How can I debug this? What to look out for? /cc @calvix @r7vme

Reply:
What about adding sleep only for workers [here](https://github.com/giantswarm/kvm-operator/blob/master/service/controller/v11/resource/deployment/worker_deployment.go#L205)

```
Command: []string{"/bin/bash", "-c", "sleep 60; /qemu-shutdown"}
```


Regarding master loses data, you need to check other resources, i bet they are still there (e.g. `kubectl get deploy -n kube-system`). Nodes are cleaned up by node-controller on purpose, because VMs are gone. But if you only recreated master then smth is broken.

Oh, wait in `geckon/gorgoth` we don't have persistent storage afaik, so it's expected that etcd data is gone.

Reply:
> Oh, wait in `geckon/gorgoth` we don't have persistent storage afaik, so it's expected that etcd data is gone.

Ok cool. This makes sense. But how can I test this with storage to make sure I do not break this somehow. AFAIK this is all fine and works well enough for now. My only worry is the master reboot and losing data.

Reply:
Adding this this sleep to `kvm-operator` repo makes sense for me in general. We could avoid some distributed knowledge.

Reply:
There will not be such delay in the end. This is just a temporary quick fix to move on.